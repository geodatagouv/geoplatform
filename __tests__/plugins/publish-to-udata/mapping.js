const mapping = require('../../../plugins/publish-to-udata/mapping')

describe('plugins.publish-to-udata.mapping', () => {
  describe('map()', () => {
    it('should map fixtures', () => {
      const fixtures = [
        'a9bcb983f29b66004c29c0cb3e0782b4cb1e663e',
        'da40697278592843273355722d08ca2db742a902'
      ]

      expect.assertions(fixtures.length)

      for (const fixture of fixtures) {
        const record = require(`./__fixtures__/mapping/${fixture}`)

        expect(mapping.map(record)).toMatchSnapshot()
      }
    })
  })
})
