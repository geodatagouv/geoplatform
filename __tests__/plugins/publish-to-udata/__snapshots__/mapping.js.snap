// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugins.publish-to-udata.mapping map() should map fixtures 1`] = `
Object {
  "description": "Couche contenant uniquement les périmètres des communes nouvelles à partir du 01/01/2016.

__Origine__

Fusion des contours de communes d&#x27;OpenStreetMap au 01/01/2015.


__Organisations partenaires__

Région Bretagne

➞ [Consulter cette fiche sur geo.data.gouv.fr](undefined/fr/datasets/a9bcb983f29b66004c29c0cb3e0782b4cb1e663e)",
  "extras": Object {
    "geop:dataset_id": "a9bcb983f29b66004c29c0cb3e0782b4cb1e663e",
    "inspire:identifier": "cc39d3e0-0ebb-4a91-8430-43a300a2b430",
    "inspire:resource_identifier": "http://applications.region-bretagne.fr/geonetwork/srv/fre/xml.metadata.get?uuid=cc39d3e0-0ebb-4a91-8430-43a300a2b430",
  },
  "license": "lov2",
  "resources": Array [
    Object {
      "description": "Conversion à la volée au format GeoJSON",
      "extras": Object {
        "geop:resource_id": "service.wfs:5bfd54c9cb932f8781e43a6e/rb:commune_nouvelle_geom",
      },
      "filetype": "remote",
      "format": "JSON",
      "title": "commune_nouvelle_geom (export GeoJSON)",
      "url": "undefined/services/5bfd54c9cb932f8781e43a6e/feature-types/rb:commune_nouvelle_geom?format=GeoJSON&projection=WGS84",
    },
    Object {
      "description": "Conversion à la volée au format Shapefile (WGS-84)",
      "filetype": "remote",
      "format": "SHP",
      "title": "commune_nouvelle_geom (export SHP/WGS-84)",
      "url": "undefined/services/5bfd54c9cb932f8781e43a6e/feature-types/rb:commune_nouvelle_geom?format=SHP&projection=WGS84",
    },
  ],
  "supplier": Object {},
  "tags": Array [
    "boundaries",
    "bretagne",
    "donnees-ouvertes",
    "communes-nouvelles",
    "unites-administratives",
    "passerelle-inspire",
  ],
  "title": "Périmètres des communes nouvelles à compter du 01/01/2016",
}
`;

exports[`plugins.publish-to-udata.mapping map() should map fixtures 2`] = `
Object {
  "description": "SRCI valide du 01/01/2016 au 31/12/2016

__Origine__

-


__Organisations partenaires__

DDT Val-d&#x27;Oise

__Liens annexes__

 * [Vue XML des métadonnées](http://ogc.geo-ide.developpement-durable.gouv.fr/csw/all-dataset?REQUEST&#x3D;GetRecordById&amp;SERVICE&#x3D;CSW&amp;VERSION&#x3D;2.0.2&amp;RESULTTYPE&#x3D;results&amp;elementSetName&#x3D;full&amp;TYPENAMES&#x3D;gmd:MD_Metadata&amp;OUTPUTSCHEMA&#x3D;http://www.isotc211.org/2005/gmd&amp;ID&#x3D;fr-120066022-jdd-b0a22b2f-7633-4049-9069-2934b6605d54)
 * [URL de base des services wms/wfs sur internet](http://ogc.geo-ide.developpement-durable.gouv.fr/wxs?map&#x3D;/opt/data/carto/geoide-catalogue/1.4/org_38124/fded01ea-3244-4444-9c55-8dfa8c6c8040.internet.map)

➞ [Consulter cette fiche sur geo.data.gouv.fr](undefined/fr/datasets/da40697278592843273355722d08ca2db742a902)",
  "extras": Object {
    "geop:dataset_id": "da40697278592843273355722d08ca2db742a902",
    "inspire:identifier": "fr-120066022-jdd-b0a22b2f-7633-4049-9069-2934b6605d54",
    "inspire:resource_identifier": "fded01ea-3244-4444-9c55-8dfa8c6c8040http://catalogue.geo-ide.developpement-durable.gouv.fr",
  },
  "license": "lov2",
  "resources": Array [
    Object {
      "filetype": "remote",
      "format": "ZIP",
      "title": "L_INTERCOMMUNALITES_095_2016.TAB (archive)",
      "url": "https://files.geo.data.gouv.fr/link-proxy/atom.geo-ide.developpement-durable.gouv.fr/2019-01-21/5c451b59154d740074ef2247/dataset-1548032856420.zip/dataset/l-intercommunalites-095-2016.zip",
    },
    Object {
      "description": "Conversion à la volée au format GeoJSON",
      "extras": Object {
        "geop:resource_id": "download:5c035cf6e21d2363215ee3b7/5c451b59154d740074ef2247",
      },
      "filetype": "remote",
      "format": "JSON",
      "title": "L_INTERCOMMUNALITES_095_2016.TAB (export GeoJSON)",
      "url": "undefined/links/5c035cf6e21d2363215ee3b7/downloads/5c451b59154d740074ef2247?format=GeoJSON&projection=WGS84",
    },
    Object {
      "description": "Conversion à la volée au format Shapefile (WGS-84)",
      "filetype": "remote",
      "format": "SHP",
      "title": "L_INTERCOMMUNALITES_095_2016.TAB (export SHP/WGS-84)",
      "url": "undefined/links/5c035cf6e21d2363215ee3b7/downloads/5c451b59154d740074ef2247?format=SHP&projection=WGS84",
    },
  ],
  "supplier": Object {},
  "tags": Array [
    "boundaries",
    "donnees-ouvertes",
    "passerelle-inspire",
  ],
  "title": "Schema départemental de cooperation intercommunale (2016)",
}
`;

exports[`plugins.publish-to-udata.mapping map() should map fixtures 3`] = `
Object {
  "description": "Zone d&#x27;application des Schéma d&#x27;Aménagement et de Gestion des Eaux sur le bassin Loire-Bretagne Centre, hors SAGE en émergence.Les Schémas d&#x27;Aménagement et de Gestion des Eaux (SAGE) fixent, dans un groupement de sous-bassins ou un sous-bassin correspondant à une unité hydrographique ou à un système aquifère, les objectifs généraux d&#x27;une gestion équilibrée de l&#x27;eau afin de satisfaire aux principes énumérés à l&#x27;article 1 de la loi sur l&#x27;eau n°92-3 du 3 janvier 1992.Objectifs : Les SAGE fixent notamment, dans le respect des orientations retenues dans le schéma directeur (SDAGE) correspondant, les objectifs de protection des écosystèmes aquatiques et des ressources en eau, d&#x27;utilisation et de mise en valeur de ces ressources et de préservation des zones humides, pour l&#x27;ensemble des ressources situées dans leurs périmètres respectifs et la totalité des usages qui y sont exercés, de manière à satisfaire les besoins présents et futurs.Procédure : L&#x27;établissement d&#x27;un SAGE comporte principalement quatre grandes phases successives :1. La phase d&#x27;émergence du SAGE2. La phase d&#x27;instruction du projet : délimitation du périmètre du SAGE fixé par arrêté préfectoral.3. La phase d&#x27;élaboration du projet : mise en place de la Commission Locale de l&#x27;Eau (CLE) et conception proprement dite du SAGE.4. La phase de mise en œuvre et de suivi du SAGE (SAGE élaboré et approuvé)



__Organisations partenaires__

DREAL Centre-Val de Loire

__Liens annexes__

 * [metadata.show](http://metadata.carmen.developpement-durable.gouv.fr/geosource/11/fre/metadata.show?uuid&#x3D;ee520540-0ad4-4d61-9cdc-4827c92a3633)

➞ [Consulter cette fiche sur geo.data.gouv.fr](undefined/fr/datasets/aa96d0041ddf5817be6e589e7ccbeb30370cd0c7)",
  "extras": Object {
    "geop:dataset_id": "aa96d0041ddf5817be6e589e7ccbeb30370cd0c7",
    "inspire:identifier": "ee520540-0ad4-4d61-9cdc-4827c92a3633",
  },
  "license": "lov2",
  "resources": Array [],
  "supplier": Object {},
  "tags": Array [
    "environment",
    "france",
    "donnees-ouvertes",
    "sage",
    "amenagement",
    "loire-bretagne",
    "passerelle-inspire",
  ],
  "title": "Schémas d'Aménagement et de Gestion des Eaux (SAGE) du bassin Loire-Bretagne",
}
`;
